apps/smtp/src/pages/api/webhooks/gift-card-sent.ts

```diff
@@ -115,44 +115,54 @@ const handler: NextWebhookApiHandler<GiftCardSentWebhookPayloadFragment> = async
-
-          return res.status(200).json({ message: "The event has been handled" });
-        },
-        (err) => {
-          const errorInstance = err[0];
-
-          if (errorInstance instanceof SendEventMessagesUseCase.ServerError) {
-            logger.error("Failed to send email(s) [server error]", { error: err });
-
-            return res.status(500).json({ message: "Failed to send email" });
-          } else if (errorInstance instanceof SendEventMessagesUseCase.ClientError) {
-            logger.info("Failed to send email(s) [client error]", { error: err });
-
-            return res.status(400).json({ message: "Failed to send email" });
-          } else if (errorInstance instanceof SendEventMessagesUseCase.NoOpError) {
-            logger.info("Sending emails aborted [no op]", { error: err });
-
-            return res.status(200).json({ message: "The event has been handled [no op]" });
-          }
-
-          logger.error("Failed to send email(s) [unhandled error]", { error: err });
-          captureException(new Error("Unhandled useCase error", { cause: err }));
-
-          return res.status(500).json({ message: "Failed to send email [unhandled]" });
-        },
-      ),
-    );
+  try {
+    return useCase
+      .sendEventMessages({
+        channelSlug: channel,
+        event: "GIFT_CARD_SENT",
+        payload,
+        recipientEmail,
+      })
+      .then((result) =>
+        result.match(
+          (r) => {
+            logger.info("Successfully sent email(s)");
+
+            return res.status(200).json({ message: "The event has been handled" });
+          },
+          (err) => {
+            const errorInstance = err[0];
+
+            if (errorInstance instanceof SendEventMessagesUseCase.ServerError) {
+              logger.error("Failed to send email(s) [server error]", { error: err });
+
+              return res.status(500).json({ message: "Failed to send email" });
+            } else if (errorInstance instanceof SendEventMessagesUseCase.ClientError) {
+              logger.info("Failed to send email(s) [client error]", { error: err });
+
+              return res.status(400).json({ message: "Failed to send email" });
+            } else if (errorInstance instanceof SendEventMessagesUseCase.NoOpError) {
+              logger.info("Sending emails aborted [no op]", { error: err });
+
+              return res.status(200).json({ message: "The event has been handled [no op]" });
+            }
+
+            logger.error("Failed to send email(s) [unhandled error]", { error: err });
+            captureException(new Error("Unhandled useCase error", { cause: err }));
+
+            return res.status(500).json({ message: "Failed to send email [unhandled]" });
+          },
+        ),
+      );
+  } catch (e) {
+    logger.error("Unhandled error from useCase", {
+      error: e,
+    });
+
+    captureException(e);
+
+    return res.status(500).send("Failed to execute");
```

Comment:
```suggestion
    return res.status(500).json({message: "SMTP app failed to handle gift card send webhook"});
```

**Suggestion:** can you send json here - it will be visible in Saleor logs as it requires JSON response